src/app/routes/article/getInitialData/addMpuBlock/index.js

```diff
@@ -0,0 +1,108 @@
+import path from 'ramda/src/path';
+import splitAt from 'ramda/src/splitAt';
+import flatten from 'ramda/src/flatten';
+import clone from 'ramda/src/clone';
+import { ARTICLE_PAGE } from '#app/routes/utils/pageTypes';
+import isLive from '#app/lib/utilities/isLive';
+
+const mpuBlock = {
+  type: 'mpu',
+  model: {},
+};
+
+const insertMpuBlock = (blocks, insertIndex) => {
+  if (!insertIndex) {
+    return blocks;
+  }
+
+  const newBlocks = clone(blocks);
+
+  const textBlocksWithParagraphs = [];
+
+  blocks.forEach((block, i) => {
+    if (block.type === 'text') {
+      const numOfParagraphs = path(['model', 'blocks'], block).filter(
+        textBlock => textBlock.type === 'paragraph',
+      ).length;
+
+      if (numOfParagraphs > 0) {
+        textBlocksWithParagraphs.push({
+          index: i,
+          numOfParagraphs,
+        });
+      }
+    }
+  });
+
+  const totalParagraphs = textBlocksWithParagraphs.reduce(
+    (acc, curr) => acc + curr.numOfParagraphs,
+    0,
+  );
+
+  // Add the MPU block after the very last block if there are not enough paragraphs
+  if (totalParagraphs < insertIndex) {
+    newBlocks.push(mpuBlock);
+    return newBlocks;
+  }
+
+  const insertData = {
+    blockIndex: null,
+    paragraphSplitAtIndex: null,
+  };
+
+  let paragraphsCount = 0;
+
+  textBlocksWithParagraphs.some(block => {
+    paragraphsCount += block.numOfParagraphs;
+
+    if (paragraphsCount >= insertIndex) {
+      insertData.blockIndex = block.index;
+      insertData.paragraphSplitAtIndex =
+        insertIndex - (paragraphsCount - block.numOfParagraphs);
+      return true;
+    }
+    return false;
+  });
+
+  const textBlock = blocks[insertData.blockIndex];
+  const textBlockParagraphs = path(['model', 'blocks'], textBlock);
+  const splitParagraphBlocks = splitAt(
+    insertData.paragraphSplitAtIndex,
+    textBlockParagraphs,
+  );
+
+  const blocksWithMpu = [
+    splitParagraphBlocks[0].length && {
+      type: 'text',
+      model: { blocks: splitParagraphBlocks[0] },
+    },
+    mpuBlock,
+    splitParagraphBlocks[1].length && {
+      type: 'text',
+      model: { blocks: splitParagraphBlocks[1] },
+    },
+  ].filter(Boolean);
+
+  newBlocks[insertData.blockIndex] = blocksWithMpu;
+
+  return flatten(newBlocks);
+};
+
+const addMpuBlock = json => {
+  // const { allowAdvertising } = path(['metadata', 'options'], json);
+  const pageType = path(['metadata', 'type'], json);
+
+  /* TODO: Remove `isLive` and replace with `allowAdvertisng` or similar when available in Ares */
+  if (isLive() || pageType !== ARTICLE_PAGE) return json;
```

Comment:
Would be worth creating a unit test to with IsLive is true. I think you've done a test when it's not an article page. 

You can put `process.env.SIMORGH_APP_ENV = 'live';` before the call (in the test) to change the env.